test/protocol_fee.test.ts

```diff
@@ -1,371 +1,155 @@
-          tickSpacing: tickSpacing,
-          newFeeReceiver: newFeeReceiver.address
-        }
-      })
-
-      const newFeeReceiverTokenXBalanceBefore = await balanceOf(
-        tokenX.contractInstance.contractId,
-        newFeeReceiver.address
-      )
-      const newFeeReceiverTokenYBalanceBefore = await balanceOf(
-        tokenY.contractInstance.contractId,
-        newFeeReceiver.address
-      )
-      const invariantTokenXBalanceBefore = await balanceOf(
-        tokenX.contractInstance.contractId,
-        invariant.address
-      )
-      const invariantTokenYBalanceBefore = await balanceOf(
-        tokenY.contractInstance.contractId,
-        invariant.address
-      )
+  test('protocol_fee', async () => {
+    const poolBefore = await getPool(invariant, tokenX, tokenY, fee, tickSpacing)
+    await withdrawProtocolFee(invariant, admin, tokenX, tokenY, fee, tickSpacing)
+    const adminParams = {
+      balanceX: await balanceOf(tokenX.contractId, admin.address),
+      balanceY: await balanceOf(tokenY.contractId, admin.address)
+    }
+    const adminExpected = {
+      balanceX: poolBefore.feeProtocolTokenX,
+      balanceY: poolBefore.feeProtocolTokenY
+    }
+    expect(adminParams).toMatchObject(adminExpected)
 
-      await WithdrawProtocolFee.execute(newFeeReceiver, {
-        initialFields: {
-          invariant: invariant.contractId,
-          token0: token0.contractInstance.contractId,
-          token1: token1.contractInstance.contractId,
-          fee: fee,
-          tickSpacing: tickSpacing
-        },
-        attoAlphAmount: MAP_ENTRY_DEPOSIT
-      })
+    const poolAfter = await getPool(invariant, tokenX, tokenY, fee, tickSpacing)
+    expect(poolAfter).toMatchObject({ feeProtocolTokenX: 0n, feeProtocolTokenY: 0n })
+  })
 
-      const newFeeReceiverTokenXBalanceAfter = await balanceOf(
-        tokenX.contractInstance.contractId,
-        newFeeReceiver.address
-      )
-      const newFeeReceiverTokenYBalanceAfter = await balanceOf(
-        tokenY.contractInstance.contractId,
-        newFeeReceiver.address
-      )
-      const invariantTokenXBalanceAfter = await balanceOf(
-        tokenX.contractInstance.contractId,
-        invariant.address
-      )
-      const invariantTokenYBalanceAfter = await balanceOf(
-        tokenY.contractInstance.contractId,
-        invariant.address
-      )
+  test('protocol fee_not_admin', async () => {
+    const notAdmin = await getSigner(ONE_ALPH * 1000n, 0)
 
-      expect(newFeeReceiverTokenXBalanceAfter).toBe(newFeeReceiverTokenXBalanceBefore + 1n)
-      expect(newFeeReceiverTokenYBalanceAfter).toBe(newFeeReceiverTokenYBalanceBefore)
-      expect(invariantTokenXBalanceAfter).toBe(invariantTokenXBalanceBefore - 1n)
-      expect(invariantTokenYBalanceAfter).toBe(invariantTokenYBalanceBefore)
-    }
+    expectError(
+      InvariantError.NotFeeReceiver,
+      invariant,
+      withdrawProtocolFee(invariant, notAdmin, tokenX, tokenY, fee, tickSpacing)
+    )
```

Comment:
There is missing case when fee receiver is changed and new receiver tries to withdraw protcol fee